tests/test_crawlera.py

```diff
@@ -51,7 +53,8 @@ class MockedEngine(object):
             def close_spider(self, spider, reason):
                 self.fake_spider_closed_result = (spider, reason)
 
-        crawler = get_crawler(spider, settings)
+        # with `spider` instead of `type(spider)` raises an exception
+        crawler = get_crawler(type(spider), settings)
```

Comment:
This change is controversial and the error it is trying to solve was introduced with Scrapy 1.7 release from last week. I think this fix deserves a PR on its own.

Reply:
The regression is caused by https://github.com/scrapy/scrapy/pull/3610

Reply:
After discussing the issue with @kmike, seems the spider instance passed to `get_crawler` was ignored anyway and a new instance was instantiated under `crawler.spider` attribute. So, this bug was already present before and should be fixed as a separate PR.